### DISCUSSION TITLE
Submit Handler Type Issue

### DISCUSSION THREAD

**Discussion by carlBgood** [react-hook-form/react-hook-form#10675] [Q&A]
Error: `Promise-returning function provided to attribute where a void return was expected.`

Using a simple form, nothing attached, no resolver. 

`onSubmit={methods.handleSubmit(onSubmit)}`

I've seen others talking about using void or a wrapper, but nothing seems to work. How was there such a breaking change in a minor update??

    type Entries = {
        email : string
    }

    const methods = useForm<Entries>()

    const onSubmit : SubmitHandler<Entries> = (data) => console.log(data)

    return (
        <FormProvider {...methods}>
            
            <form onSubmit={methods.handleSubmit(onSubmit)}>
                <input {...methods.register("email")} />
                <input type="submit" />
            </form>
            
        </FormProvider>
    )

**Comment by leapful** [comment]
@carlBgood 

There is no breaking change in react-hook-form. The error that you got is caused by ESLint rule `no-misused-promises` , you can apply this option to bypass the error: https://typescript-eslint.io/rules/no-misused-promises/#checksvoidreturn-true

Note: In the future, when you found any issue of any library or framework which you thought it was a breaking change, please also specify the old good version number and new version number, maybe a codesandbox url to re-produce would be way more better. Just asking why there is a breaking change in a "minor update" does help to investigate the problem.
